sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -239,7 +248,11 @@ public void finishBundle() {
     if (state == State.UNSTARTED) {
       startBundle();
     }
-    fnRunner.finishBundle();
+    try {
+      fn.finishBundle(createContext(fn));
+    } catch (Exception e) {
+      throw UserCodeException.wrap(e);
```

Comment:
This is approximately equal to
```
try {
  fn.startBundle();
} catch (Exception e) {
  throw new RuntimeException(e);
}
```

as `DoFn#processElement()` is declared to throw an exception. Alternatively, this method could be declared to throw an exception (which is probably reasonable), although it puts a slightly higher burden on test writers.